src/include/daos_fs.h

```diff
@@ -519,9 +519,13 @@ dfs_mkdir(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode,
  * \param[in]	parent	Opened parent directory object. If NULL, use root obj.
  * \param[in]	name	Name of object to remove in parent dir.
  * \param[in]	force	If true, remove dir even if non-empty.
- * \param[in]	oid	Optionally return the DAOS Object ID of the removed obj.
+ * \param[in,out]
+ *		oid	[in]: Only delete this object
```

Comment:
oh i see you changing this here now.. 
but his is still not clear as even IN should be optional. I'd rather you revert this change to the API and we can discuss / handle in a different PR.
The original behavior is it's an optional out parameter only (typo in the param setting, but the description is correct). if you want to pass in the oid to verify that is changing the behavior of the API and requires and API minor version bump.

Reply:
Ok, I can put this in a different PR.

The issue is that the permissions check has already been done by dfuse using the xattr based permission bits so dfs doesn't handle multiple users and doesn't know the users anyway, making oid an IN allows us to verify that the file deleted is the one we intended.